src/main/java/com/google/devtools/build/lib/skyframe/ProcessPackageDirectory.java

```diff
@@ -187,6 +178,35 @@ public ProcessPackageDirectoryResult getPackageExistenceAndSubdirDeps(
         /*additionalValuesToAggregate=*/ ImmutableMap.of());
   }
 
+  // Note that it's not enough to just check for the convenience symlinks themselves,
+  // because if the value of --symlink_prefix changes, the old symlinks are left in place. It
+  // is also not sufficient to check whether the symlink points to a directory in the current
+  // exec root, since this can change between bazel invocations. Therefore we check if the
+  // suffix of the symlink source suggests it is a convenience symlink, then see if the symlink
+  // target is in a directory that looks like an execroot. This algorithm also covers more
+  // creative use cases where people create convenience symlinks somewhere in the directory
+  // tree manually.
+  private boolean isConvenienceSymlink(FileValue fileValue, PathFragment rootRelativePath) {
+    if (!fileValue.isSymlink()) {
+      return false;
+    }
+
+    PathFragment linkTarget = fileValue.getUnresolvedLinkTarget();
+
+    return linkTarget.startsWith(directories.getExecRootBase().asFragment())
+      || (rootRelativePath.getBaseName().endsWith("-bin") && isInExecRoot(linkTarget, 4))
```

Comment:
nit: also add `-genfiles`

Reply:
Do you know what depth this is at? I've never seen this before (is it just a blaze thing?)

Reply:
Yep. You can treat it exactly the same way as `-bin`, it's a symlink to one of its siblings that we haven't gotten around removing yet.